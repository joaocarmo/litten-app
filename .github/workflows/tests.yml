# This workflow will do a clean install of node dependencies, build the source
# code and run the available tests
name: Tests

on:
  push:
    branches: [ main, next ]
  pull_request:
    branches: [ main, next ]

env:
  CI: true
  FIREBASE_EMULATORS_PATH: $GITHUB_WORKSPACE/emulator-cache

jobs:
  test:
    environment: development
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - name: Use Node.js v16 (LTS)
      uses: actions/setup-node@v3
      with:
        node-version: '16'
        cache: 'yarn'
    - name: Install firebase-tools globally
      run: yarn global add firebase-tools
    - name: Create .env file
      run: cp .env.example .env
    - name: Install dependencies
      run: yarn --frozen-lockfile
    - name: Lint the source files
      run: yarn lint
    - name: Type check
      run: yarn type-check
    - name: Validate the translations
      run: yarn translations-check
    - name: Execute the tests
      run: yarn test
    - name: Execute the firebase rules tests
      run: yarn test:firestore:rules
    - name: Execute the firebase model tests
      run: yarn test:firestore:model
    - name: Cache firebase emulators
      uses: actions/cache@v3
      with:
        path: ${{ env.FIREBASE_EMULATORS_PATH }}
        key: ${{ runner.os }}-firebase-emulators-${{ hashFiles('emulator-cache/*.{jar,zip}') }}
      continue-on-error: true
