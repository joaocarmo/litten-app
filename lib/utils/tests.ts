import { apps, clearFirestoreData } from '@firebase/rules-unit-testing'

export const projectId = 'litten-app'

export const cleanFirestore = async (): Promise<void> => {
  await Promise.all(apps().map((app) => app.delete()))
  await clearFirestoreData({
    projectId,
  })
}

export const mapName =
  <T>(key: string) =>
  (object: T): T =>
    Object.assign(object, {
      toString: function () {
        return this[key]
      },
    })

export const getRandInt = (min: num, max: num): num =>
  Math.floor(Math.random() * (max - min + 1) + min)
