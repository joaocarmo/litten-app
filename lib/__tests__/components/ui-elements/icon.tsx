import { render } from '@utils/tests/react-native'
import { UIIcon } from '@ui-elements'

describe('Snapshot test for the "UIText" component', () => {
  it('renders correctly', () => {
    const element = render(
      <UIIcon
        icon={{
          uri: '',
        }}
        elevated
        circle
      />,
    )

    expect(element.toJSON()).toMatchSnapshot()
  })
})
