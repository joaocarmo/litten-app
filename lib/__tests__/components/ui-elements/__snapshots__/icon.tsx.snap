// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the "UIText" component renders correctly 1`] = `
<View
  elevated={true}
  style={
    Array [
      Object {
        "aspectRatio": 1,
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
      },
      Object {
        "height": 40,
        "padding": 10,
      },
      Object {
        "borderRadius": 20,
      },
      Object {
        "elevation": 8,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 6,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "height": "100%",
            "tintColor": "#39a4f2",
            "width": "100%",
          },
          undefined,
          undefined,
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="contain"
      source={
        Object {
          "uri": "",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
